src/testers.jl

```diff
@@ -244,6 +244,17 @@ function _ensure_not_running_on_functor(f, name)
     end
 end
 
+"""
+    @maybe_inferred f(...)
```

Comment:
Can we think of a better name for this?
Might not be able to.

Reply:
`maybe_test_inferred` is perhaps more accurate?

Reply:
I wish `@inferred` has been named `@test_inferred` (and that it returnned a bool rather than erroring)

Reply:
I agree that it's not the greatest name, but I couldn't really think of a better one. Since it isn't an actual test I don't think it should have `test` in its name, but I am open to any other suggestions.

Reply:
We could make it an actual test

Reply:
I think I'd prefer to make it consistent with base, so it's just a drop-in replacement for `@inferred`. We could think about having an `@maybe_test_inferred` as well, but for now this seems more straightforward to me.